src/Container/EmitterFactory.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Expressive\Container;
+
+use Psr\Container\ContainerInterface;
+use Zend\Diactoros\Response\EmitterInterface;
+use Zend\Diactoros\Response\SapiEmitter;
+use Zend\Expressive\Emitter\EmitterStack;
+
+class EmitterFactory
+{
+    public function __invoke(ContainerInterface $container) : EmitterInterface
```

Comment:
I would go with `EmitterStack` as return type (unless we'd like to have this class extensible and return another implementation of `EmitterInterface`).

Just thinking - maybe we should have `final` factories?

Reply:
Again, same reasons as for `ApplicationPipelineFactory`. While an `EmitterStack` is the default, the idea is to note that any emitter is valid for this purpose.

In terms of `final` factories: sure, we could go that way. If we do, I'll update the typehints to the more specific classes returned.